doc/api/inspector.md

```diff
@@ -11,86 +11,207 @@ inspector.
-* `wait` {boolean} Block until a client has connected. Optional.
-  **Default:** `false`.
+<!-- YAML
+added: v8.0.0
+-->
 
-Activate inspector on host and port. Equivalent to
-`node --inspect=[[host:]port]`, but can be done programmatically after node has
-started.
+* {Object} The notification message object
 
-If wait is `true`, will block until a client has connected to the inspect port
-and flow control has been passed to the debugger client.
+Emitted when an inspector notification is received that has its method field set
+to the `<inspector-protocol-method>` value.
 
-See the [security warning][] regarding the `host`
-parameter usage.
+The following snippet installs a listener on the [`'Debugger.paused'`][]
+event, and prints the reason for program suspension whenever program
+execution is suspended (through breakpoints, for example):
 
-## `inspector.url()`
+```js
+session.on('Debugger.paused', ({ params }) => {
+  console.log(params.hitBreakpoints);
+});
+// [ '/the/file/that/has/the/breakpoint.js:11:0' ]
+```
 
-* Returns: {string|undefined}
+#### `session.connect()`
 
-Return the URL of the active inspector, or `undefined` if there is none.
+<!-- YAML
+added: v8.0.0
+-->
 
-```console
-$ node --inspect -p 'inspector.url()'
-Debugger listening on ws://127.0.0.1:9229/166e272e-7a30-4d09-97ce-f1c012b43c34
-For help, see: https://nodejs.org/en/docs/inspector
-ws://127.0.0.1:9229/166e272e-7a30-4d09-97ce-f1c012b43c34
+Connects a session to the inspector back-end.
 
-$ node --inspect=localhost:3000 -p 'inspector.url()'
-Debugger listening on ws://localhost:3000/51cf8d0e-3c36-4c59-8efd-54519839e56a
-For help, see: https://nodejs.org/en/docs/inspector
-ws://localhost:3000/51cf8d0e-3c36-4c59-8efd-54519839e56a
+#### `session.connectToMainThread()`
 
-$ node -p 'inspector.url()'
-undefined
-```
+<!-- YAML
+added: v12.11.0
+-->
+
+Connects a session to the main thread inspector back-end. An exception will
+be thrown if this API was not called on a Worker thread.
 
-## `inspector.waitForDebugger()`
+#### `session.disconnect()`
 
 <!-- YAML
-added: v12.7.0
+added: v8.0.0
 -->
 
-Blocks until a client (existing or connected later) has sent
-`Runtime.runIfWaitingForDebugger` command.
+Immediately close the session. All pending message callbacks will be called
+with an error. [`session.connect()`][] will need to be called to be able to send
+messages again. Reconnected session will lose all inspector state, such as
+enabled agents or configured breakpoints.
 
-An exception will be thrown if there is no active inspector.
+#### `session.post(method[, params])`
+
+<!-- YAML
+added: REPLACEME
+-->
 
-## Class: `inspector.Session`
+* `method` {string}
+* `params` {Object}
+* Returns: {Promise}
+
+Posts a message to the inspector back-end.
+
+```mjs
+import { Session } from 'inspector/promises';
+try {
+  const session = new Session();
+  session.connect();
+  const result = await session.post('Runtime.evaluate', { expression: '2 + 2' });
```

Comment:
**Question:** would it be worth mentioning that the way internally it works is that instead of passing a callback by using promises, the "result" of the `session.post` is still a deferred callback? 

(I'm assuming that `Promise.all([s.post(1), s.post(2), s.post(3)]` still entails that these promises not necessarily will be resolved in that order as the execution to them is deferred? When I mean not in that order, I'm referring to that post(1) might resolve after post(3))

That might be worth mentioning, as you're not blocking your execution thread when you're using callbacks. Still, there will be a difference in how your execution thread will behave by either using Promises or async/await.

**Note.:** Please ignore if it doesn't apply to this scenario; I'm assuming the internals of the inspector API as I haven't _really_ looked deep into its inner workings.

Reply:
Well, honestly I haven't seen a problem while promisifying a function and resulting in a wrong order. I did the test case just in case. Would you check it?

Reply:
> Well, honestly I haven't seen a problem while promisifying a function and resulting in a wrong order. I did the test case just in case. Would you check it?

Oh, I think you got me wrong the result order should be still the same, I was just referring that the order of resolution might not be the same.